Remove `AStar2D/3D` comments on `reserve_space()` capacity needs

smix8 opened the pull request:
Removes AStar2D/3D comments of `reserve_space()` new capacity needing to be greater or equal than old capacity.

This is no longer an error case with recent PR https://github.com/godotengine/godot/pull/105278 merged.

Documentation could still merge https://github.com/godotengine/godot/pull/102941 so users know the default point alloc.

<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

RedMser reviewed (commented):
This should update the method documentation to remove this sentence:

"The new capacity must be greater or equal to the old capacity."

Or rewrite it to "will always be", or something even more explicit...

Ivorforce commented:
I already have https://github.com/godotengine/godot/pull/105278 open, which would also solve this problem :)

smix8 committed:
Remove AStar2D/3D comments on reserve_space() capacity needs

Removes AStar2D/3D comments of reserve_space() new capacity needing to be greater or equal than old capacity.

Repiteo commented:
Thanks!
